devimint/src/external.rs

```diff
@@ -237,13 +239,53 @@ impl Bitcoind {
         Ok(proof.encode_hex())
     }
 
-    pub async fn get_raw_transaction(&self, txid: bitcoin::Txid) -> Result<String> {
+    /// Get a transaction by its txid. Checks the mempool and all blocks.
+    pub async fn get_transaction(&self, txid: bitcoin::Txid) -> Result<String> {
+        // Check the mempool.
+        if let Ok(tx) = self.get_raw_transaction(txid, None).await {
+            return Ok(tx);
+        };
+
+        let block_height = self.get_block_count().await? - 1;
+
+        let mut per_block_check_futures = FuturesUnordered::new();
```

Comment:
Even in a medium size test, this can lead to tens of thousands of futures, rpc calls and possibly start hitting some limits.

Reply:
I'm thinking ... I would change this to `buffered` stream going in reverse from the highest height to the lowest with some windows like 128...

Reply:
```
futures::stream::iter((0..block_height).rev()).map(...).buffered(128)
```

Reply:
Good call! Done.